spec/controllers/groups_controller_spec.rb

```diff
@@ -479,6 +484,98 @@
 
         expect(response.parsed_body).to eq expected
       end
+
+      describe 'when users are already in groups' do
+        let!(:grouping) { create :grouping_with_inviter, assignment: assignment, inviter: student1 }
+        let(:assignment2) { create :assignment }
+        let(:grouping2) { create :grouping_with_inviter, assignment: assignment2, inviter: student2 }
+        it 'does not match a student already in a grouping' do
+          post_as instructor, :get_names, params: { course_id: course.id,
+                                                    assignment_id: assignment.id,
+                                                    assignment: assignment.id,
+                                                    term: '123',
+                                                    format: :json }
+          expect(response.parsed_body).to be_empty
+        end
+        it 'does match students not in the group but in other assignment groups' do
+          post_as instructor, :get_names, params: { course_id: course.id,
+                                                    assignment_id: assignment.id,
+                                                    assignment: assignment.id,
+                                                    term: '789',
+                                                    format: :json }
+
+          expect(response.parsed_body).to eq [{ 'id' => student2.id,
+                                                'id_number' => student2.id_number,
+                                                'user_name' => student2.user_name,
+                                                'value' => "#{student2.first_name} #{student2.last_name}" }]
+        end
+      end
+      describe 'when multiple students match' do
+        let(:expected) do
+          [{ 'id' => student2.id,
+             'id_number' => student2.id_number,
+             'user_name' => student2.user_name,
+             'value' => "#{student2.first_name} #{student2.last_name}" },
+           { 'id' => student3.id,
+             'id_number' => student3.id_number,
+             'user_name' => student3.user_name,
+             'value' => "#{student3.first_name} #{student3.last_name}" }]
+        end
+        it 'returns multiple matches' do
+          post_as instructor, :get_names, params: { course_id: course.id,
+                                                    assignment_id: assignment.id,
+                                                    assignment: assignment.id,
+                                                    term: 'zz',
+                                                    format: :json }
+
+          expect(response.parsed_body).to eq expected
```

Comment:
Avoid using `eq` for arrays because order isn't guaranteed from database queries. Instead, use [`match_array`](https://www.rubydoc.info/gems/rspec-expectations/2.99.2/RSpec%2FMatchers:match_array).